cip25/rust/src/utils.rs

```diff
@@ -109,6 +109,87 @@ impl From<&ChunkableString> for String {
+    pub image: Option<ChunkableString>,
+}
+
+impl MiniMetadataDetails {
+    pub fn new(name: Option<String64>, image: Option<ChunkableString>) -> Self {
+        Self {
+            name,
+            image,
+        }
+    }
+
+    /// loose parsing of CIP25 metadata to allow for common exceptions to the format
+    /// TODO: this is not an ideal solution
+    ///       ideally: we would have a function that takes in a policy ID
+    ///       and would have a lookup map to know which lambda to call to get the name & image depending on the policy ID
+    ///       with a fallback to the standard CIP25 definition
+    ///       however, since this is a lot of work, we use this temporary solution instead
+    pub fn loose_parse(data: Vec<u8>) -> Result<Self, DeserializeError> {
+        let metadatum = TransactionMetadatum::from_bytes(
+            data,
+        )?;
+        match metadatum {
+            TransactionMetadatum::Map { entries, .. } => {
+                let name: Option<String64> = entries
+                .get(&TransactionMetadatum::new_text("name".to_owned()))
+                .or_else(|| entries.get(&TransactionMetadatum::new_text("name".to_owned())))
+                // for some reason, 1% of NFTs seem to use the wrong case
+                .or_else(|| entries.get(&TransactionMetadatum::new_text("Name".to_owned())))
+                // for some reason, 0.5% of NFTs use "title" instead of name
+                .or_else(|| entries.get(&TransactionMetadatum::new_text("title".to_owned())))
+                // for some reason, 0.3% of NFTs use "id" instead of name
+                .or_else(|| entries.get(&TransactionMetadatum::new_text("id".to_owned())))
+                .map(|result| match result {
+                    TransactionMetadatum::Text { text, .. } => String64::new_str(&text).ok(),
+                    _ => None,
+                })
+                .flatten();
+    
+                let image_base = entries
+                .get(&TransactionMetadatum::new_text("image".to_owned()));
+                let image = match image_base {
+                    None => None,
+                    Some(base) => match base {
+                        TransactionMetadatum::Text { text, .. } => match String64::new_str(&text) {
+                            Ok(str64) => Some(ChunkableString::Single(str64)),
+                            Err(_) => None,
+                        },
+                        TransactionMetadatum::List { elements, .. } => {
+                            (|| {
+                                let mut chunks: Vec<String64> = vec![];
+                                for i in 0..elements.len() {
+                                    match elements.get(i) {
+                                        Some(TransactionMetadatum::Text { text, .. }) => {
+                                            match String64::new_str(&text) {
+                                                Ok(str64) => chunks.push(str64),
```

Comment:
I feel like maybe `ChunkableString ` and similar types should be provided as utility types in some common crate since this affects all metadata definitions for Cardano (same as String64, etc)
For now, I'll just inline this here